records/generator.go

```diff
@@ -134,12 +134,21 @@ func (rg *RecordGenerator) loadWrap(ip string, port string) (StateJSON, error) {
 	return sj, err
 }
 
-// yankPort grabs the first port in the port field
-// this takes a string even though it should take an array
-func yankPort(ports string) string {
+// yankPorts takes an array of port ranges
+// FIXME - does not use ranges
+func yankPorts(ports string) []string {
 	rhs := strings.Split(ports, "[")[1]
 	lhs := strings.Split(rhs, "]")[0]
-	return strings.Split(lhs, "-")[0]
+
+	yports := []string{}
+
+	mports := strings.Split(lhs, ", ")
```

Comment:
Would it not be better to split on `,` and then lstrip/rstrip any extra whitespace?